readthedocs/search/models.py

```diff
@@ -81,10 +82,109 @@ def generate_queries_count_of_one_month(cls, project_slug):
 
         # format the date value to a more readable form
         # Eg. `16 Jul`
-        last_31_days_str = [
-            timezone.datetime.strftime(date, '%d %b')
-            for date in last_31_days_iter
-        ]
+        last_31_days_str = _get_last_31_days_str(date_format='%d %b')
+
+        final_data = {
+            'labels': last_31_days_str,
+            'int_data': count_data,
+        }
+
+        return final_data
+
+
+class PageView(models.Model):
```

Comment:
We should consider having a [unique index](https://docs.djangoproject.com/en/1.11/ref/models/options/#unique-together) for this model although it could result in a lot of space used. Nonetheless, it would prevent a whole class of problem:

```python
class Meta:
    unique_together = ("project", "version", "path", "date")
```